### PR TITLE
Cloud Connect Identification Logic Fix

### LINKED ISSUE #121: Current logic incomplete for identifying Cloud Connect
### Bug description

Cloud Connect is currently identified using this logic ([link](https://github.com/AsBuiltReport/AsBuiltReport.Veeam.VBR/blob/dev/Src/Private/Get-AbrVbrCloudConnectTenant.ps1#L29)): 

`Get-VBRInstalledLicense | Where-Object {$_.CloudConnect -in @("Enterprise")}`

However, this logic does not work for service providers using Cloud Connect. Instead, the below logic should be used:

`Get-VBRInstalledLicense | Where-Object {$_.CloudConnect -ne "Disabled"}`

### Command-line input

New-AsBuiltReport -Report Veeam.VBR -Target $HostName -Credential $Credential -Format HTML -OutputFolderPath $Desktop -EnableHealthCheck -ReportConfigFilePath $ReportConfigFile -AsBuiltConfigFilePath $ConfigFile

### Steps to reproduce

1. Run script against Cloud Connect server using Veeam service provider license.
2. Check generated report
3. Report does not show detailed Cloud Connect information

### Expected behaviour

As long as correct InfoLevel is defined in `AsBuiltReport.Veeam.VBR.json`, Cloud Connect reports should work for both Enterprise and Service Providers.

### Screenshots

CleanShot 2023-12-29 at 10 42 29

### Operating System

Server 2022

### PowerShell Version

CleanShot 2023-12-29 at 10 43 05

### PowerShell Modules

CleanShot 2023-12-29 at 10 43 24

### Additional Context

_No response_

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://www.asbuiltreport.com/about/contributing/#reporting-issues-and-bugs).
- [X] I have read [the documentation](https://www.asbuiltreport.com/user-guide/new-asbuiltconfig), and referred to the [known issues](https://www.asbuiltreport.com/user-guide/known-issues/) before submitting this bug report.
- [X] I have checked for previously opened & closed [issues](https://github.com/AsBuiltReport/AsBuiltReport.Veeam.VBR/issues) before submitting this bug report.

### PR DESCRIPTION
**Pull request by carceneaux**
## Description
Cloud Connect is currently identified using this logic ([link](https://github.com/AsBuiltReport/AsBuiltReport.Veeam.VBR/blob/dev/Src/Private/Get-AbrVbrCloudConnectTenant.ps1#L29)): 

`Get-VBRInstalledLicense | Where-Object {$_.CloudConnect -in @("Enterprise")}`

However, this logic does not work for service providers using Cloud Connect. Instead, the below logic should be used:

`Get-VBRInstalledLicense | Where-Object {$_.CloudConnect -ne "Disabled"}`

## Related Issue

Fixes #121 

## Motivation and Context

Current logic is unable to determine if Cloud Connect for Service Providers is being used. As such, all of the amazing Cloud Connect reporting is missing from the report.

## How Has This Been Tested?

Tested in two separate lab environments where a Cloud Connect server was being used with a service provider Cloud Connect license installed.

Each lab environment is unique and is configured and managed by different people. A lab environment consists of a functioning Cloud Connect environment.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.

### CODE REVIEW

File: Src/Private/Get-AbrVbrCloudConnectRR.ps1 [hunk 1]
```diff
@@ -6,7 +6,7 @@ function Get-AbrVbrCloudConnectRR {
     .DESCRIPTION
         Documents the configuration of Veeam VBR in Word/HTML/Text formats using PScribo.
     .NOTES
-        Version:        0.7.1
+        Version:        0.7.2
```
Comment (rebelinux) [inline Src/Private/Get-AbrVbrCloudConnectRR.ps1:9]:
```suggestion
        Version:        0.8.3
```

File: Src/Private/Get-AbrVbrCloudConnectStatus.ps1 [hunk 2]
```diff
@@ -6,7 +6,7 @@ function Get-AbrVbrCloudConnectStatus {
     .DESCRIPTION
         Documents the configuration of Veeam VBR in Word/HTML/Text formats using PScribo.
     .NOTES
-        Version:        0.7.1
+        Version:        0.7.2
```
Comment (rebelinux) [inline Src/Private/Get-AbrVbrCloudConnectStatus.ps1:9]:
```suggestion
        Version:        0.8.3
```

File: Src/Private/Get-AbrVbrCloudConnectTenant.ps1 [hunk 3]
```diff
@@ -6,7 +6,7 @@ function Get-AbrVbrCloudConnectTenant {
     .DESCRIPTION
         Documents the configuration of Veeam VBR in Word/HTML/Text formats using PScribo.
     .NOTES
-        Version:        0.8.0
+        Version:        0.8.1
```
Comment (rebelinux) [inline Src/Private/Get-AbrVbrCloudConnectTenant.ps1:9]:
```suggestion
        Version:        0.8.3
```

File: Src/Public/Invoke-AsBuiltReport.Veeam.VBR.ps1 [hunk 4]
```diff
@@ -5,7 +5,7 @@ function Invoke-AsBuiltReport.Veeam.VBR {
     .DESCRIPTION
         Documents the configuration of Veeam VBR in Word/HTML/Text formats using PScribo.
     .NOTES
-        Version:        0.8.3
+        Version:        0.8.4
```
Comment (rebelinux) [inline Src/Public/Invoke-AsBuiltReport.Veeam.VBR.ps1:8]:
```suggestion
        Version:        0.8.3
```

**Review by rebelinux — CHANGES_REQUESTED** [review]
Thanks for the fixes!

File: AsBuiltReport.Veeam.VBR.psd1 [hunk 5]
```diff
@@ -12,7 +12,7 @@
 RootModule = 'AsBuiltReport.Veeam.VBR.psm1'
 
 # Version number of this module.
-ModuleVersion = '0.8.3'
+ModuleVersion = '0.8.4'
```
[the remark this answers is not in the document]
Comment (carceneaux) [reply AsBuiltReport.Veeam.VBR.psd1:15]: I've applied all other requested changes.

Don't we need to increment the module version so PowerShell gallery is updated?

[the remark this answers is not in the document]
Comment (rebelinux) [reply AsBuiltReport.Veeam.VBR.psd1:15]: the latest PowerShell gallery version is v0.8.2:

https://www.powershellgallery.com/packages/AsBuiltReport.Veeam.VBR/0.8.2

dev version is v0.8.3, if everything is working fine, I intend to release that version before 2024.

**Review by rebelinux — APPROVED** [review]

**Pull request merged**